interp/src/flatten/structures/environment/program_counter.rs

```diff
@@ -56,6 +56,67 @@ impl ControlPoint {
             false
         }
     }
+
+    /// Returns a string showing the path from the root node to input node.
+    /// How to get context?
+    pub fn string_path(&self, ctx: &Context) -> String {
+        // Does it matter if it takes ownership?
+        let path = SearchPath::find_path_from_root(self.control_node_idx, ctx);
+        let control_map = &ctx.primary.control;
+        let mut string_path = String::from("");
+        let mut count = -1;
```

Comment:
Prefer `new` over `from` with an empty `&str`
```suggestion
        let mut string_path = String::new();
```

Reply:
I guess my suggestion thing is just cursed, this also belongs on the line above